components/brave_wallet/browser/zcash/zcash_complete_transaction_task.cc

```diff
@@ -0,0 +1,321 @@
+                             &consensus_branch_id)) {
+    error_ = l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR);
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  consensus_branch_id_ = consensus_branch_id;
+  transaction_.set_consensus_brach_id(consensus_branch_id);
+  ScheduleWorkOnTask();
+}
+
+void ZCashCompleteTransactionTask::GetLatestBlock() {
+  context_.zcash_rpc->GetLatestBlock(
+      context_.chain_id,
+      base::BindOnce(&ZCashCompleteTransactionTask::OnGetLatestBlockHeight,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void ZCashCompleteTransactionTask::OnGetLatestBlockHeight(
+    base::expected<zcash::mojom::BlockIDPtr, std::string> result) {
+  if (!result.has_value()) {
+    error_ = l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR);
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  chain_tip_height_ = result.value()->height;
+
+  transaction_.set_locktime(result.value()->height);
+  transaction_.set_expiry_height(result.value()->height +
+                                 kDefaultZCashBlockHeightDelta);
+  ScheduleWorkOnTask();
+}
+
+#if BUILDFLAG(ENABLE_ORCHARD)
+
+void ZCashCompleteTransactionTask::GetMaxCheckpointedHeight() {
+  CHECK(chain_tip_height_);
+  if (transaction_.orchard_part().inputs.empty()) {
+    anchor_block_height_ = chain_tip_height_;
+    ScheduleWorkOnTask();
+    return;
+  }
+  context_.sync_state->AsyncCall(&OrchardSyncState::GetMaxCheckpointedHeight)
+      .WithArgs(context_.account_id.Clone(), chain_tip_height_.value(),
+                kMinConfirmations)
+      .Then(base::BindOnce(
+          &ZCashCompleteTransactionTask::OnGetMaxCheckpointedHeight,
+          weak_ptr_factory_.GetWeakPtr()));
+}
+
+void ZCashCompleteTransactionTask::OnGetMaxCheckpointedHeight(
+    base::expected<std::optional<uint32_t>, OrchardStorage::Error> result) {
+  if (!result.has_value() || !result.value()) {
+    error_ = l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR);
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  anchor_block_height_ = *result;
+  ScheduleWorkOnTask();
+}
+
+void ZCashCompleteTransactionTask::CalculateWitness() {
+  if (transaction_.orchard_part().inputs.empty()) {
+    witness_inputs_ = std::vector<OrchardInput>();
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  context_.sync_state
+      ->AsyncCall(&OrchardSyncState::CalculateWitnessForCheckpoint)
+      .WithArgs(context_.account_id.Clone(), transaction_.orchard_part().inputs,
+                anchor_block_height_.value())
+      .Then(base::BindOnce(
+          &ZCashCompleteTransactionTask::OnWitnessCalulcateResult,
+          weak_ptr_factory_.GetWeakPtr()));
+}
+
+void ZCashCompleteTransactionTask::OnWitnessCalulcateResult(
+    base::expected<std::vector<OrchardInput>, OrchardStorage::Error> result) {
+  if (!result.has_value()) {
+    error_ = l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR);
+    ScheduleWorkOnTask();
+    return;
+  }
+
+  // TODO(cypt4): rewrite
```

Comment:
nit: is there an issue for this?

Reply:
Honestly i don't remember what was the idea, so just removed this